### PR TITLE
Different bad steps

### PR DESCRIPTION
**Pull request by RyanYoung25**
Adds in 4 new types of bad steps that mentii can make when solving algebra problems. 

**Significant Changes**
Added: 
  - Swap signs
  - Swap operators
  - Swap numbers
  - Drop last term
  - Add new term

**How to Test**
1. Do lots of problems and look at the types of bad steps that you see

**Notes**
If you guys during testing notice any crazy bad steps let me know and we can discuss what we should do if anything.

### CODE REVIEW

**Comment by smc395** [comment]
So I've been testing this manually and I think the changes so far make the experience way more realistic in my opinion. :1st_place_medal: :100: .

The only thing that I don't get it the addition of random numbers or terms somewhere. I guess I can't see a student in real life that Mentii is supposed to mimicking just adding a +6 to the equation out of no where. Also I know it's difficult (and you can totally not do it if you don't have enough time or someone could try pick it up), but it would be good if the +x's in problems were just x.

**Comment by asp78** [comment]
Ditto Micah stuff, I'd like adding random stuff dropped, or even better swapped for the "incorrect" operations where Mentii would make a mistake in adding two things. That's a common mistake that isn't in here yet

**Comment by smc395** [comment]
mentii

I thought the system only swaps "-" when there is a space after?

File: Backend/problems/algebra.py [hunk 1]
```diff
@@ -34,19 +35,109 @@ def getProblem(problemTemplate):
...
-  elif(badStep.find('/') != -1):
-    badStep = badStep.replace('/', "*")
-  elif(badStep.find('+') != -1):
-    badStep = badStep.replace('+', "-")
-  elif(badStep.find('-') != -1):
-    badStep = badStep.replace('-', "+")
+  success = False
+  containsNegOne = r'-1(?:[^\d]|$)'
+  if not re.search(containsNegOne, step):
+    modifications = [dropTerm, swapSign, swapOperator, swapNumbers, newTerm]
```
Comment (smc395) [inline Backend/problems/algebra.py]: can we take out newTerm per my comments? unless you think otherwise

**Pull request merged**